### PR TITLE
Subject badge- input handling

### PR DESCRIPTION
**Pull request by desmatter**
Tested it locally and it was bugging with improper input, due to the sujects being hardcoded. Refactored if statement to account for this.

### CODE REVIEW

File: bin/cron.js [hunk 1]
```diff
@@ -157,7 +163,14 @@ cron.schedule("*/15 * * * *", async () => {
                   if (submission.score >= 90) {
```
Comment (jnoowin) [inline bin/cron.js:163]: I missed this in the previous pr. The application quizzes don't have a score minimum on then so there doesn't need ot be a requirement for it to be above 90. Ideally, it should use the actual minimum stored in mongo but we can implement that later.

File: bin/cron.js [hunk 2]
```diff
@@ -157,7 +163,14 @@ cron.schedule("*/15 * * * *", async () => {
...
                     score += 100;
                     completed.apply += 1;
-                    if (subj && userProgress.modules[moduleID].practice) {
+                    if (
+                      (subj === "chem" ||
+                        subj === "econ" ||
+                        subj === "physics" ||
+                        subj === "engineering" ||
+                        subj === "bio") &&
+                      userProgress.modules[moduleID].practice
```
Comment (jnoowin) [inline bin/cron.js:172]: Why are the application subject quizzes dependent on if the practice has been completed? I kinda see why as the applications should always be taken after the practice ones, but I'm not sure about having it be a requirement in the code. Let me know what you think

Comment (desmatter) [reply bin/cron.js:172]: The badge req has the condition that the ALEKS(practice) is complete.

File: bin/cron.js (hunk 1 above)
Comment (desmatter) [reply bin/cron.js:163]: I saw that yea that some badges just have "Application Quiz Attempted", wasn't sure. But I agree

File: bin/cron.js (hunk 2 above)
Comment (jnoowin) [reply bin/cron.js:172]: I think I might have made a mistake with the other pr. The sole purpose of this function here should be to count the number of assignments types completed, separate from any badge requirements. The order the submissions come in isn't ordered in any way, so with the way the code is right now, an application quiz that come first won't get counted if the practice one comes after it. They won't get awarded the badge then when they should have. The logic for if they get awarded a badge should be in updateBadgeProgress in canvas.js.

**Review by jnoowin — CHANGES_REQUESTED** [review]

**Pull request merged**